pkg/cmd/roachtest/tests/logical_data_replication.go

```diff
@@ -220,6 +234,80 @@ func TestLDRBasic(ctx context.Context, t test.Test, c cluster.Cluster, setup mul
 	VerifyCorrectness(t, setup, leftJobID, rightJobID, 2*time.Minute, ldrWorkload)
 }
 
+func TestLDRSchemaChange(
+	ctx context.Context, t test.Test, c cluster.Cluster, setup multiClusterSetup,
+) {
+	duration := 15 * time.Minute
+	if c.IsLocal() {
+		duration = 5 * time.Minute
+	}
+
+	ldrWorkload := LDRWorkload{
+		workload: replicateKV{
+			readPercent:             0,
+			debugRunDuration:        duration,
+			maxBlockBytes:           1024,
+			initRows:                1000,
+			initWithSplitAndScatter: true},
+		dbName:    "kv",
+		tableName: "kv",
+	}
+
+	leftJobID, rightJobID := setupLDR(ctx, t, c, setup, ldrWorkload)
+
+	// Setup latency verifiers
+	maxExpectedLatency := 2 * time.Minute
+	llv := makeLatencyVerifier("ldr-left", 0, maxExpectedLatency, t.L(),
+		getLogicalDataReplicationJobInfo, t.Status, false /* tolerateErrors */)
+	defer llv.maybeLogLatencyHist()
+
+	rlv := makeLatencyVerifier("ldr-right", 0, maxExpectedLatency, t.L(),
+		getLogicalDataReplicationJobInfo, t.Status, false /* tolerateErrors */)
+	defer rlv.maybeLogLatencyHist()
+
+	// Run allowlisted schema changes on the replicated table on both the left
+	// and right sides.
+	setup.left.sysSQL.Exec(t, fmt.Sprintf("ALTER TABLE %s.%s ADD COLUMN new_col INT NULL", ldrWorkload.dbName, ldrWorkload.tableName))
```

Comment:
i feel like we should test the allow list with unit tests first and merge a roachtest later? See `logical_replication_job_test.go` for examples on setting up a unit test.

Reply:
thanks, that's the pointer i needed! i can also add logic tests that are SQL only